torch/_functorch/aot_autograd.py

```diff
@@ -1039,9 +1060,16 @@ def inner(*flat_args):
                     mutates_metadata = has_metadata_mutation(f_arg)
                 # Only track requires_grad info on *mutated* inputs,
                 # because they show up in the autograd.Function.forward as outputs
-                input_requires_grad_info.append(
-                    isinstance(f_arg, torch.Tensor) and f_arg.requires_grad
-                )
```

Comment:
Why did this change?

If this is the thing where "input mutations on subclasses never go in the graph in the training case", I don't think we should change the requires_grad field here. Instead, in `ViewAndMutationMeta.__post_init__()` we should just directly use `self.is_subclass`

Reply:
Some inputs don't show up as forward outputs i think.